scan

```diff
@@ -431,12 +439,23 @@ def perform_lambda_scan(scanner, domain, handles, environment, options, meta):
 
         response = json.loads(raw)
 
-        if response.get("errorMessage") is None:
+        if response is None:
+            meta['errors'].append("Response came back empty. Raw payload response:\n%s\nFull api_response:\n%s" % (raw, api_response))
+
+        # An errorMessage field implies a Lambda-level error.
+        elif response.get("errorMessage") is None:
             # Payload has some per-task Lambda-specific info.
             meta['lambda'] = {**meta['lambda'], **response['lambda']}
 
-            # Payload has the actual scan response data.
-            data = response['data']
+            if 'data' in response:
+                # Payload has the actual scan response data.
+                data = response['data']
+            else:
+                meta['errors'].append("Response object lacked 'data' field. Raw response: %s" % raw)
+
+            # An error field implies an exception during the scan.
+            if 'error' in response:
+                meta['errors'].append("Error or exception during scan: %s" % response['error'])
```

Comment:
Testing out the new Node-based Lambda handler/functions also led me to harden the general error-handling paths for Lambda function execution. We were getting local errors and stack traces in handling Lambda function responses, which are now handled gracefully and return context-specific information for easier debugging.

This also results in us being able to see Node stack traces in the returned `--meta` information we save to scan CSVs.